pkg/app/client/option.go

```diff
@@ -147,3 +148,34 @@ func WithWriteTimeout(t time.Duration) config.ClientOption {
 		o.WriteTimeout = t
 	}}
 }
+
+// WithDialFunc is used to set dialer function for DefaultDialer.
+// Deprecated: use WithDialer instead
```

Comment:
If decide to provide this api, remove this deprecated tag.

Reply:
done